Add support for ESP32-{S2,S3,C3} to debug component

jayme-github opened the pull request:
# What does this implement/fix?

Add support for ESP32-{S2,S3,C3} to debug component.

## Types of changes

Not sure if it should be classified as feature instead.

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes [ esphome/issues/3433
](https://github.com/esphome/issues/issues/3433)
## Test Environment

- [x] ESP32
- [ ] ESP32 IDF
- [ ] ESP8266
- [ ] RP2040


## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

jayme-github committed:
Add support for ESP32-{S2,S3,C3} to debug component

Include variant specific rtc.h headers and update reset_reason accordingly.

Fixes esphome/issues/3433

nielsnl68 commented:
Hi, can you make sure the test issues are resolved? Then we can do a review.

nielsnl68 reviewed (approved):
Looks good to me.

nielsnl68 commented:
With the new options you added, should there be no new entries for the `sensor` and `text_sensor` platform components.

jayme-github commented:
> With the new options you added, should there be no new entries for the `sensor` and `text_sensor` platform components.

I'm not sure what you are referring to. I'm not really experienced with ESP things and/or this codebase so I might be missing something obvious. My understanding was that this only supports proper reset reasons (text sensor values) or those particular ESP32 variants.

kahrendt reviewed (commented):
I have tested this on an S3, and it works well! The only issue is that it states the chip is unknown. In the switch at line [120](https://github.com/esphome/esphome/pull/4731/files#diff-9a34821728cbc6e3297a26de0565002449519327fbb1622ca76fecc7e8649540R120), you should add the new cases (see [espressif's documentation](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/system/misc_system_api.html#_CPPv416esp_chip_model_t))

For example, I used this to get the proper chip model for my S3 device:
```
  switch (info.model) {
    case CHIP_ESP32:
      model = "ESP32";
      break;
    case CHIP_ESP32S3:
      model = "ESP32-S3";
      break;
    default:
      model = "UNKNOWN";
  }
```

jayme-github committed:
Update model info for C3/S2/S3/H2

Update model info for C3/S2/S3/H2 chips

jesserockz commented on esphome/components/debug/debug_component.cpp:
```diff
@@ -113,6 +121,18 @@ void DebugComponent::dump_config() {
     case CHIP_ESP32:
       model = "ESP32";
       break;
+    case CHIP_ESP32C3:
+      model = "ESP32-C3";
+      break;
+    case CHIP_ESP32S2:
+      model = "ESP32-S2";
+      break;
+    case CHIP_ESP32S3:
+      model = "ESP32-S3";
+      break;
+    case CHIP_ESP32H2:
+      model = "ESP32-H2";
+      break;
```
Actually one change to make here would be to wrap each of these in `#if defined(USE_ESP32_VARIANT_ESPxx)` like the includes as we know the chip cant be the others when compiling.

jayme-github replied:
I agree. Sorry that it took me so long to come around again.

jayme-github committed:
Replace model switch with ifdefs

The ESP model will only be one of the variant when compiling. No need to
check at runtime.

jayme-github committed:
Merge branch 'dev' into issue3433

jayme-github committed:
Merge branch 'dev' into issue3433
